include/phasar/PhasarLLVM/DataFlow/IfdsIde/Problems/IDEInstInteractionAnalysis.h

```diff
@@ -1044,55 +1001,15 @@ class IDEInstInteractionAnalysisT
       return IDEInstInteractionAnalysisT::joinImpl(Src, Data);
     }
 
-    static EdgeFunction<l_t> compose(EdgeFunctionRef<IIAAAddLabelsEF> This,
-                                     const EdgeFunction<l_t> &SecondFunction) {
-      if (auto Default = defaultComposeOrNull(This, SecondFunction)) {
-        return Default;
-      }
-
-      auto Cache = This.getCacheOrNull();
-      assert(Cache != nullptr && "We expect a cache, because "
-                                 "IIAAAddLabelsEF is too large for SOO");
-
-      if (auto *AD = llvm::dyn_cast<IIAAAddLabelsEF>(SecondFunction)) {
-        auto Union =
-            IDEInstInteractionAnalysisT::joinImpl(This->Data, AD->Data);
-        return Cache->createEdgeFunction(std::move(Union));
-      }
-      if (auto *KR = llvm::dyn_cast<IIAAKillOrReplaceEF>(SecondFunction)) {
-        return SecondFunction;
-      }
-      llvm::report_fatal_error(
-          "found unexpected edge function in 'IIAAAddLabelsEF'");
+    static EdgeFunction<l_t>
+    compose(EdgeFunctionRef<IIAAAddLabelsEF> /*This*/,
+            const EdgeFunction<l_t> & /*SecondFunction*/) {
+      llvm::report_fatal_error("Implemented in 'extend'");
```

Comment:
Is this left for legacy reasons?

Reply:
Currently, the `EdgeFunction` interface requires `compose` and `join` to be implemented. The default implementation of the semi-ring methods also assumes these functions to be implemented. We may want to make them optional at some point

Reply:
If they not strictly required, disabling them would be a option, otherwise this could crash at some point.